managed/CounterStrikeSharp.API/Modules/Memory/DynamicFunctions/BaseMemoryFunction.cs

```diff
@@ -45,6 +47,30 @@ private static IntPtr CreateValveFunctionBySignature(string signature, string bi
         }
 
         return function;
+    }
+
+    private static IntPtr CreateValveFunctionByOffset(IntPtr objectPtr, int offset, DataType returnType,
+        DataType[] argumentTypes)
+    {
+        if (!_createdOffsetFunctions.TryGetValue(objectPtr, out var createdFunctions))
+        {
+            createdFunctions = new Dictionary<int, IntPtr>();
+            _createdOffsetFunctions[objectPtr] = createdFunctions;
+        }
+
+        if (!createdFunctions.TryGetValue(offset, out var function))
+        {
+            try
+            {
+                function = NativeAPI.CreateVirtualFunction(objectPtr, offset,
+                    argumentTypes.Length, (int)returnType, argumentTypes.Cast<object>().ToArray());
+                createdFunctions[offset] = function;
+            } catch (Exception)
+            {
+            }
+        }
+
+        return function;
```

Comment:
might need a better way? if we pass another entity instance (ptr) with the same offset it will probably try to hook the same function again and fail? (untested currently, will answer here when I can test it)